### PR TITLE
workflows: Extend IPsec tests to cover egress gateway

### PR DESCRIPTION
**Pull request by pchaigno**
The first two commits are cleanup for the IPsec workflows. The third extends coverage to IPsec + KPR + egress gateway. The last one avoids the matrix deduplication and prettyfies the job display.

### CODE REVIEW

**Review by aanm — APPROVED** [review]
@pchaigno shall we backport this PR?

File: .github/workflows/tests-ipsec-upgrade.yaml [hunk 1]
```diff
@@ -100,95 +100,70 @@ jobs:
       fail-fast: false
       max-parallel: 16
       matrix:
-        config: ['5.4', '5.10', '6.1', '5.15', 'bpf-next']
+        name: ['1', '2', '3', '4', '5', '6', '7', '8']
```
Comment (aanm) [inline .github/workflows/tests-ipsec-upgrade.yaml]: I just noticed the output of the tests results, can we give them a more descriptive name? https://github.com/cilium/cilium/actions/runs/11332423439

At least something similar as https://github.com/cilium/cilium/actions/runs/11332421340

**Review by aanm — CHANGES_REQUESTED** [review]

**Review by jschwinger233 — APPROVED** [review]

**Review by nebril — APPROVED** [review]

**Review by aanm — APPROVED** [review]

**Comment by julianwiedmann** [comment]
To be robust, I suspect this will need the same `--expected-drop-reasons=+No egress gateway found` toleration as in `ci-e2e-upgrade`.

**Comment by julianwiedmann** [comment]
> @pchaigno shall we backport this PR?

I believe we should, adding the label.

**Comment by pchaigno** [comment]
> > @pchaigno shall we backport this PR?
> 
> I believe we should, adding the label.

I don't think we can. IPsec+KPR is not supported in v1.15 so upgrade workflow would fail.

**Comment by julianwiedmann** [comment]
> > > @pchaigno shall we backport this PR?
> > 
> > 
> > I believe we should, adding the label.
> 
> I don't think we can. IPsec+KPR is not supported in v1.15 so upgrade workflow would fail.

Let's try something funky, how do you feel about https://github.com/cilium/cilium/pull/35540 ?

File: .github/actions/ipsec/configs.yaml [hunk 2]
```diff
@@ -0,0 +1,86 @@
+- name: '1'
+  # renovate: datasource=docker depName=quay.io/lvh-images/kind
+  kernel: 'rhel8-20240404.144247'
+  kube-proxy: 'iptables'
```
Comment (julianwiedmann) [inline .github/actions/ipsec/configs.yaml:4]: Note that this file is currently not in scope of our renovate config, so we'll no longer auto-bump the LVH images.

Comment (pchaigno) [reply .github/actions/ipsec/configs.yaml:4]: Thanks! Fix is at https://github.com/cilium/cilium/pull/35555.

**Pull request merged**